app/gui/view/src/project/dashboard_button.rs

```diff
@@ -0,0 +1,121 @@
+//! Provides a button to switch back from the project view to the dashboard.
+
+use crate::prelude::*;
+
+use ensogl::application::tooltip;
+use ensogl::application::Application;
+use ensogl::display;
+use ensogl::display::shape::StyleWatch;
+use ensogl::display::Scene;
+use ensogl_component::toggle_button;
+use ensogl_component::toggle_button::ToggleButton;
+use ensogl_hardcoded_theme::application::dashboard_button as theme;
+
+
+
+// ============
+// === Icon ===
+// ============
+
+/// Defines an icon for returning to the dashboard. It looks like a hamburger button.
+mod icon {
+    use super::*;
+
+    use ensogl::data::color;
+
+    use ensogl_component::toggle_button::ColorableShape;
+
+    ensogl::shape! {
+        alignment = center;
+        (style: Style, color_rgba: Vector4<f32>) {
+            let fill_color = Var::<color::Rgba>::from(color_rgba);
+            let width = Var::<Pixels>::from("input_size.x");
+            let height = Var::<Pixels>::from("input_size.y");
+            let unit = &width / 16.0;
+            let mid_bar = Rect((&unit * 12.0, &unit * 3.0)).corners_radius(&unit);
+            let top_bar = mid_bar.translate_y(&unit * -5.0);
+            let bottom_bar = mid_bar.translate_y(&unit * 5.0);
+            let all_bars = top_bar + mid_bar + bottom_bar;
+            let shape = all_bars.fill(fill_color);
+            let hover_area = Rect((&width, &height)).fill(INVISIBLE_HOVER_COLOR);
+            (shape + hover_area).into()
+        }
+    }
+
+    impl ColorableShape for Shape {
+        fn set_color(&self, color: color::Rgba) {
+            self.color_rgba.set(Vector4::new(color.red, color.green, color.blue, color.alpha));
+        }
+    }
+}
+
+
+
+// ============
+// === Frp  ===
+// ============
+
+ensogl::define_endpoints! {
+    Input {
+    }
+    Output {
+        pressed (),
+        size (Vector2<f32>),
+    }
+}
+
+
+
+// ============
+// === View ===
+// ============
+
+/// Provides a button to switch back from the project view to the dashboard.
+#[derive(Debug, Clone, CloneRef)]
+pub struct View {
+    button:  ToggleButton<icon::Shape>,
+    #[allow(missing_docs)]
+    pub frp: Frp,
+}
```

Comment:
Wait, but my question here is why – were there any reason for that? Doing that the way you did requires a lot of boilerplate, that is simply not needed if we reuse an existing component. So my question here is if there was a reason to produce so much boilerplate. If there was a reason, let's chat about it, maybe some API needs to be improved.

Reply:
Oh, sure, my initial reasoning was that it's nice for someone who reads the code to only have to jump one level down to see the API of the dashboard button instead of having to realise that it's actually a toggle button and then jumping down another level to find the API for toggle buttons. I don't think it matters much though.